### PR TITLE
feat: onboarding ui flow

### PR DESCRIPTION
**Pull request by magrinj**
https://github.com/twentyhq/twenty/assets/3551795/aeed9ea5-c34a-45ec-83c8-5d1e9e124f95

### CODE REVIEW

**Review by lucasbordeau — COMMENTED** [review]
- Nice job with the animations
- Code is very clean
- Are we sure we want to introduce function Component and const Component at the same time in the codebase now ? Since we still have to decide.
- Are we sure that we want to allow drill-down of props on every component ? We haven't decided this yet.
- Why using interface & types in props ? It introduces another pattern
- Using styled component directly with props is again another pattern, quite interesting though.
- There's a lot of memoization on tiny functions, I'm not sure that the performance gain is worth it. Maybe it's worth talking about this also, because it's not a react native project. I like this article from Kent C Dodds about this exact discussion we have here : https://kentcdodds.com/blog/usememo-and-usecallback

**Comment by FelixMalfait** [comment]
Nice 🤩 !!!

Not for this PR but as we add more uploads across the app in the future maybe it'll be useful to have a shared component that has a drag-n-drop behaviour

(Ex: [react-dropzone.js.org](https://react-dropzone.js.org/#section-basic-example))

File: infra/dev/package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "framer-motion": "^10.12.17"
```
[the remark this answers is not in the document]
Comment (magrinj) [reply infra/dev/package.json:3]: Oops ! 😅

**Comment by magrinj** [comment]
> * Nice job with the animations
> * Code is very clean
Thanks ! 🙂

> * Are we sure we want to introduce function Component and const Component at the same time in the codebase now ? Since we still have to decide.
I've removed all the `React.FC` for now and used `function` instead of `arrow function` for the component I've created.

> * Are we sure that we want to allow drill-down of props on every component ? We haven't decided this yet.
This is very useful in many case but we can discuss that yes

> * Why using interface & types in props ? It introduces another pattern
`interface` are often used to define props of a component, and type is more used when you need to create complex type like unions. Also interface allow extension with `extends`.
https://levelup.gitconnected.com/mastering-typescript-a-guide-to-choosing-between-type-and-interface-c31d3527693b#:~:text=In%20conclusion%2C%20both%20type%20and,complex%20types%20in%20your%20props.

> * Using styled component directly with props is again another pattern, quite interesting though.
It's often use when you need to create complex style based on the props of your component, by exemple in the `Button` and `MainButton` component we can't get really rid of that.

> * There's a lot of memoization on tiny functions, I'm not sure that the performance gain is worth it. Maybe it's worth talking about this also, because it's not a react native project. I like this article from Kent C Dodds about this exact discussion we have here : https://kentcdodds.com/blog/usememo-and-usecallback
I've removed useful ones

**Comment by charlesBochet** [comment]
@magrinj nice one! Can we merge it? :)

**Review by charlesBochet — APPROVED** [review]
Thanks a lot for this PR and all the interesting discussions!

**Pull request merged**